Resign all add-ons with new COSE signature.

EnTeQuAk opened the pull request:
This is re-using the old sign_addons task but renames the argument
needed for `process_addons --task=` command so that we make sure the new
code won't be used by accident by some cron or so.

* Cleans up some old jetpack/legacy add-on related code along the way

Fixes #11323

diox commented on src/olympia/lib/crypto/tasks.py:
```diff
@@ -8,68 +8,37 @@
 
-If you are unfamiliar with the extension signing requirement, please read the
-following documents:
+Mozilla has recently upgraded the signing for Firefox extensions and themes to
```
`extensions` is probably wrong here, that will affect themes and langpacks as well, right ? This is why the original message said `add-on`.

EnTeQuAk replied:
It's saying "extensions and themes" but you're right, language packs are affected as well. cc @caitmuenster on this

diox replied:
Wow I'm blind. The problem is present on the line below though, it says `all extension versions`.

EnTeQuAk replied:
dictionaries are affected as well imho

EnTeQuAk replied:
Question is, does "Extension" in a wider sense include themes and the others too? Not sure about our general wording in this regard.

diox replied:
Historically it didn't, at least...

caitmuenster replied:
Good points & thanks for the catch! Made a few tweaks here: https://docs.google.com/document/d/167MXie1DEXDVEM0j2V8ALC8MBSjjKqNp6Ebd9TV9gMg/edit#

diox commented on src/olympia/lib/crypto/tasks.py:
```diff
@@ -91,49 +60,58 @@ def get_new_version_number(version):
+    current_versions = (
+        Addon.objects
+        .filter(id__in=addon_ids)
+        .exclude(type=amo.ADDON_SEARCH)
```
Shouldn't this be done earlier, in `process_addons` task declaration ?

EnTeQuAk replied:
Actually, I just re-read the `process-addon` command and I think you're right, all that should be moving there. I'll fix that.

diox commented on src/olympia/lib/crypto/tasks.py:
```diff
@@ -91,49 +60,58 @@ def get_new_version_number(version):
+        .values_list('_current_version', flat=True))
+    qset = (
+        Version.objects.filter(id__in=current_versions)
+        .exclude(addon__type=amo.ADDON_SEARCH))
```
Shouldn't be needed.

EnTeQuAk commented:
except for the wording question this should be ready for another review.

diox commented on src/olympia/addons/management/commands/process_addons.py:
```diff
@@ -41,7 +41,10 @@
         'method': sign_addons,
-        'qs': []},
+        'qs': [
+            ~Q(type=amo.ADDON_SEARCH)
```
Now that I think about it we probably want to restrict to public add-ons only. No point in re-signing others. The task will filter some of the non-public out because it looks at `current_version`, but that's not enough.

EnTeQuAk commented:
* Updated messaging and added test
* We're now only re-signing add-ons created before April 5th since we introduced the new signing on April 4th
* Only public add-ons are being considered

re-r? @diox

diox commented on src/olympia/lib/crypto/tests/test_signing.py:
```diff
@@ -451,213 +451,188 @@ def test_sign_full(self, mock_sign_file):
+            self.version.reload()
+            assert self.version.version == '0.0.1'
+            assert self.version.version_int == version_int('0.0.1')
+            assert file_hash != new_file.generate_hash()
```
This doesn't make sense, it has nothing to do with `self.version`. Wrong copy/paste ?

EnTeQuAk replied:
Fixed in 63f046794a, indeed, that didn't make any sense :roll_eyes: yeah, some copy&paste error :-/
